logging.go

```diff
@@ -147,3 +151,140 @@ func (lr *LoggingResource) ValidateAWSTrustPolicy(ctx context.Context, awsExtern
+	Logged          time.Time      `json:"logged"`                    // the time at which this log was captured by the server
+	NodeID          string         `json:"nodeId"`                    // the node ID for which the flow statistics apply
+	Start           time.Time      `json:"start"`                     // the start of the sample period (node's local clock)
+	End             time.Time      `json:"end"`                       // the end of the sample period (node's local clock)
+	VirtualTraffic  []TrafficStats `json:"virtualTraffic,omitempty"`  // traffic between Tailscale nodes
+	SubnetTraffic   []TrafficStats `json:"subnetTraffic,omitempty"`   // traffic involving subnet routes
+	ExitTraffic     []TrafficStats `json:"exitTraffic,omitempty"`     // traffic via exit nodes
+	PhysicalTraffic []TrafficStats `json:"physicalTraffic,omitempty"` // WireGuard transport-level statistics
+}
+
+// TrafficStats represents traffic flow statistics.
+// This type is used for all traffic types: virtual, subnet, exit, and physical.
+type TrafficStats struct {
+	Proto   int    `json:"proto,omitempty"`   // IP protocol number (e.g., 6 for TCP, 17 for UDP)
+	Src     string `json:"src,omitempty"`     // Source address and port
+	Dst     string `json:"dst,omitempty"`     // Destination address and port
+	TxPkts  uint64 `json:"txPkts,omitempty"`  // Transmitted packets
+	TxBytes uint64 `json:"txBytes,omitempty"` // Transmitted bytes
+	RxPkts  uint64 `json:"rxPkts,omitempty"`  // Received packets
+	RxBytes uint64 `json:"rxBytes,omitempty"` // Received bytes
+}
+
+// NetworkFlowLogsRequest represents query parameters for fetching network flow logs.
+type NetworkFlowLogsRequest struct {
+	// Start must be set to a non-zero time within the log retention period (last 30 days).
+	// The server may adjust times that are too old.
+	Start time.Time
+	// End must be set to a non-zero time after Start.
+	End time.Time
+}
+
+// NetworkFlowLogHandler is a callback function for processing individual network flow log entries.
+// It receives each log entry as it's parsed from the JSON stream.
+// Return an error to stop processing and bubble up the error.
+type NetworkFlowLogHandler func(log NetworkFlowLog) error
+
+// GetNetworkFlowLogs streams network flow logs for the tailnet, calling the provided
+// handler function for each log entry as it's parsed from the JSON response.
+// This approach is memory-efficient and handles large datasets without loading all logs into memory.
+//
+// Both start and end parameters are required by the server.
+// Times older than 30 days will be automatically adjusted by the server to the retention limit.
+func (lr *LoggingResource) GetNetworkFlowLogs(ctx context.Context, params NetworkFlowLogsRequest, handler NetworkFlowLogHandler) error {
+
+	url := lr.buildTailnetURL("logging", "network")
+	q := url.Query()
+	q.Set("start", params.Start.Format(time.RFC3339))
+	q.Set("end", params.End.Format(time.RFC3339))
+	url.RawQuery = q.Encode()
```

Comment:
Optional simplification, now that we aren't interleaving checks, if you want:
```suggestion
	url.RawQuery = url.Values{
		"start": {params.Start.Format(time.RFC3339)},
		"end":   {params.End.Format(time.RFC3339)},
	}.Encode()
```

but it's up to you.